lib/graphlient/client.rb

```diff
@@ -16,28 +16,28 @@ def parse(&block)
         instance_eval(&block)
       end
       client.parse(query_str.to_s)
+    rescue GraphQL::Client::Error => e
+      raise Graphlient::Errors::Client.new(e.message, e)
```

Comment:
Ruby automatically captures nested exceptions and this logic seems redundant. See [this ruby spec](https://github.com/ruby/spec/blob/f0548a9a0bb7271f334a30d7c34dbd56a8764189/core/exception/cause_spec.rb). Also, the name `Client` doesn't imply that it's an error class. I would call it `ClientError` instead.

Reply:
That was renamed to `ClientError` in #32, thanks. 

We want to rename the exception from the underlying library into something higher level from this library.